Backpack-SwiftUI/ChipGroup/Classes/BPKSingleSelectChipGroup.swift

```diff
@@ -0,0 +1,146 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+/// A Group of chips that allows a single chip to be selected at a time.
+public struct BPKSingleSelectChipGroup: View {
+    private let chips: [ChipItem]
+    private let style: BPKChipStyle
+    private let type: BPKChipGroupType
+    private let onItemClick: (_ index: Int) -> Void
+    
+    @Binding private var selectedIndex: Int?
```

Comment:
made this a Binding, so that we move the 'holder' of this state to being the client, instead of the component itself. Clients can now change the value of the `selectedIndex` on their side and this would update automatically as well!

Reply:
Also made it optional, to allow for the 'no chip selected' case